pg_shard.c

```diff
@@ -741,14 +894,170 @@ RowAndColumnFilterQuery(Query *query, List *remoteRestrictList, List *localRestr
 	fromExpr->quals = (Node *) make_ands_explicit((List *) remoteRestrictList);
 	fromExpr->fromlist = QueryFromList(rangeTableList);
 
+	if (remoteQueryHasAggregates)
+	{
+		distributedQuery->hasAggs = true;
+		distributedQuery->groupClause = list_copy(query->groupClause);
+		distributedQuery->targetList =
+			BuildAggregatedDistributedTargetList(query, localRestrictList);
+
+		/*
+		 * Note that havingQual is in implicitly-ANDed-list form
+		 * at this point, even though they are declared as Node.
+		 */
+		distributedQuery->havingQual =
+			(Node *) make_ands_explicit((List *) query->havingQual);
+	}
+	else
+	{
+		distributedQuery->targetList =
+			BuildNonAggregatedDistributedTargetList(query, localRestrictList);
+	}
+
+	distributedQuery->commandType = CMD_SELECT;
+	distributedQuery->rtable = rangeTableList;
+	distributedQuery->jointree = fromExpr;
+
+	return distributedQuery;
+}
+
+
+/*
+ * BuildAggregatedDistributedTargetList returns a list of TargetEntry for the
+ * distributed query when aggregates are pushed down. This function leaves
+ * AggRefs as it is in the target list. Also, WHERE clause columns pushed down.
+ */
+static List *
+BuildAggregatedDistributedTargetList(Query *query, List *localRestrictList)
```

Comment:
`Aggregated` is ambiguous. Let's go with `AggregateBearing` for queries containing aggregates and `Aggregateless` for queries without them.